app/controllers/api/v1/users_controller.rb

```diff
@@ -27,13 +27,13 @@ def create
         token: auth_params["access_token"],
         refresh_token: auth_params["refresh_token"]
       )
-      session[:user_id] = @user.id
       redirect_to "/api/v1/preferences"
     else
       @user.update_attributes(token: auth_params["access_token"], refresh_token: auth_params["refresh_token"])
-      session[:user_id] = @user.id
       redirect_to '/api/v1/dashboard'
     end
+    session[:user_id] = @user.id
```

Comment:
Please explain this to me when you have a chance. If I understand this correctly, you've created a temporary session that exists along side the usual session[:user_id] so that you can then change it later in the artists index page.